File: app_dagger2_sample_03_mvp/src/main/java/com/rsh_engineering/tkachenkoni/app_dagger2_sample_03_mvp/presentation/MainActivity.java
package com.rsh_engineering.tkachenkoni.app_dagger2_sample_03_mvp.presentation;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.rsh_engineering.tkachenkoni.app_dagger2_sample_03_mvp.App04;
import com.rsh_engineering.tkachenkoni.app_dagger2_sample_03_mvp.R;

/**
 * Created by Nikolay Tkachenko on 02, October, 2020
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((App04)getApplication()).getComponent().inject(this);


    }
}

File: app/src/main/java/com/rsh_engineering/tkachenkoni/daggersampleexample/presentation/MainActivity.kt
package com.rsh_engineering.tkachenkoni.daggersampleexample.presentation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.rsh_engineering.tkachenkoni.daggersampleexample.App
import com.rsh_engineering.tkachenkoni.daggersampleexample.R
import com.rsh_engineering.tkachenkoni.daggersampleexample.app
import com.rsh_engineering.tkachenkoni.daggersampleexample.data_domain.LoginManager
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    //Use debug logcat and key "DAGGER_USE_TEST"

    @Inject
    lateinit var loginManager : LoginManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /*
        //Use injection without dagger
        val localStore = LocalStore("test store")
        val apiService = ApiService("nikolay","12345")
        val manager = LoginManager(localStore, apiService)

        loginManager = loginComponent.getLoginManager()
        loginManager?.login("nikolay", "12345")
         */

        //Use injection use Dagger
        val loginComponent = applicationContext.app.loginComponent
        loginComponent.injectMainActivity(this)

        loginManager.login("nikolay", "12345")

    }

}